Userland/Libraries/LibIMAP/Parser.cpp

```diff
@@ -270,7 +270,12 @@ ErrorOr<void> Parser::parse_untagged()
 ErrorOr<StringView> Parser::parse_quoted_string()
 {
     dbgln_if(IMAP_PARSER_DEBUG, "p: {}, parse_quoted_string()", m_position);
-    auto str = consume_while([](u8 x) { return x != '"'; });
+    int chars = 0;
+    while (!at_end() && (m_buffer[m_position] == '"' ? m_buffer[m_position - 1] == '\\' : true)) {
```

Comment:
That condition looks a) inverted (…right?) b) overly complicated. I think you want:

```
while (!at_end() && (m_buffer[m_position] != '"' || m_buffer[m_position - 1] == '\\')) {
```

Please give `chars` a more descriptive name, e.g. `number_of_characters`, `string_length`, or similar. Alternatively, do `auto start_position = m_position;` before the loop, and `auto str = StringView(m_buffer.data() + start_position, m_position - start_position);` after the loop.

Reply:
> That condition looks a) inverted (…right?)

I don't think so, it's the same condition I was passing to `consume_while`, however 

> b) overly complicated. I think you want:

Agreed.

> Alternatively, do auto start_position = m_position; before the loop, and auto str = StringView(m_buffer.data() + start_position, m_position - start_position); after the loop.

Done